tendrl/notifier/notification/__init__.py

```diff
@@ -120,8 +120,9 @@ def _run(self):
                     alert.tags = json.loads(alert.tags)
                     if alert.delivered == "False":
                         for plugin in NotificationPlugin.plugins:
-                            plugin.dispatch_notification(alert)
-                        update_alert_delivery(alert)
+                            err = plugin.dispatch_notification(alert)
+                        if not err:
```

Comment:
This line would capture the last error only. Shouldn't we handle each error within for loop?

Reply:
mm ya i got it